### ISSUE TITLE
Terminal detection failed in Linux Mint 18 under SMAPI 3.0

### ISSUE THREAD

**Issue by techge** [Pathoschild/SMAPI#676]
The game (SDV 1.4.1) doesn't start, in terminal you only see
```
$ ./start.sh                                                                                                   
Running Stardew Valley                                                                                                                                 
command                                                                                                                                                
/usr/bin/x-terminal-emulator
```

Thus, the terminal detection in `game/StardewValley` apparently does not work anymore (did in SMAPI 2). After installing xterm it worked again as it falls back to this in case of problems.

This means, installing xterm is a workaround, but there seems to be a problem detecting installed terminals underlying.

**Comment by Pathoschild** [comment]
Terminal compatibility is the eternal struggle with supporting Linux; we've been iterating on the launcher script for almost two years. At this point I'm tempted to just add 'install xterm' as a step in the Linux install guide.

**Comment by Pathoschild** [comment]
Can you come ask in [#modding on the Stardew Valley Discord](https://smapi.io/community) so we can run some tests? I just found the fix for an issue that affects Arch Linux in SMAPI 3.0.0; hopefully that'll fix it for you too, or if not we'll run some tests to track down the issue.

**Comment by techge** [comment]
I would prefer to rather get some diff and test it than joining the Discord because this way I can test when I have time to do so. The linux mint system is someone else's system, not mine. I am using Arch :wink:

**Comment by Pathoschild** [comment]
Sure. First, can you see if [the latest launcher](https://github.com/Pathoschild/SMAPI/blob/develop/src/SMAPI.Installer/unix-launcher.sh) for the upcoming release works?

**Comment by techge** [comment]
> Sure. First, can you see if the latest launcher for the upcoming release works?

Thanks, but it does not help (I did apply the change and removed xterm -> did not work anymore).

The change itself looks totally fine to me though. Maybe I just try to add the terminal used in Mint (not sure if it is Mate or Cinnamon, will have a look) to the list of known-to-work terminals and check if it *does* work. What do you think?

**Comment by Pathoschild** [comment]
Sure. The launch script only needs to handle preferred terminals, or terminals which don't work correctly through `x-terminal-emulator`. Based on the output above, looks like whichever terminal is being used is the latter.

**Comment by Pathoschild** [comment]
Fixed in #678.
